src/db/spanner/batch.rs

```diff
@@ -355,23 +355,28 @@ pub async fn do_append_async(
         } else {
             let sortindex = bso
                 .sortindex
-                .map(|sortindex| as_value(sortindex.to_string()))
+                .map(|sortindex| sortindex.to_spanner_value())
+                .unwrap_or_else(null_value);
+            let payload = bso
+                .payload
+                .as_ref()
+                .map(ToSpannerValue::to_spanner_value)
                 .unwrap_or_else(null_value);
-            let payload = bso.payload.map(as_value).unwrap_or_else(null_value);
             let ttl = bso
                 .ttl
-                .map(|ttl| as_value(ttl.to_string()))
+                .as_ref()
+                .map(ToSpannerValue::to_spanner_value)
                 .unwrap_or_else(null_value);
 
             // convert to a protobuf structure for direct insertion to
             // avoid some mutation limits.
             let mut row = ListValue::new();
             row.set_values(RepeatedField::from_vec(vec![
-                as_value(user_id.fxa_uid.clone()),
-                as_value(user_id.fxa_kid.clone()),
-                as_value(collection_id.to_string()),
```

Comment:
There were a few cases like this where I removed the call to `to_string()` and just converted directly to a Value from `i32`/`f32`. From what I could tell, this seemed to be okay/desirable, but I wasn't 100% sure, so I thought I'd mention it here

Reply:
We might want to check on this, because javascript is kinda notorious at screwing up parsing of numerics. Converting to string at least normalizes them.